### PR TITLE
Restyle headers a bit

### PR DESCRIPTION
**Pull request by pavpanchekha**
This PR makes three big changes to styling. In each pair of screenshots below, I show "before" first, than "after".

# First: move the chapter numbers on the main page so they're within the page bounds. This makes them visible on mobile.

Before:
image

After:
Screenshot 2024-06-06 at 3 50 56 PM

# Second: Use CSS counters to number sections, instead of Pandoc. This allows us to style the section number:

Before:
Screenshot 2024-06-06 at 3 51 51 PM

After:
image

Also we don't need to put the counter into the table of contents:

Before:
image

After:
Screenshot 2024-06-06 at 3 52 54 PM

# Third: Use a pseudoelement "#" instead of an underline to suggest clicking a header for a permalink. This has bothered me for a while, the underlines look bad.

Before:
image

After:
Screenshot 2024-06-06 at 3 54 00 PM

### CODE REVIEW

File: www/book.css [hunk 1]
```diff
@@ -48,10 +54,10 @@ a:active { color: #f64; }
 
 li { margin: 1em 0; }
 
-.main ol, .main > h1 { font-size: 120%; }
+.main ol, .main h1 { font-size: 120%; }
```
Comment (chrishtr) [inline www/book.css:57]: Why is this change necessary?

File: www/book.css [hunk 2]
```diff
@@ -31,9 +31,15 @@ header, h1, strong {
 }
 }
 h1 { font-size: 105%; margin: 1.5em 0 0; }
+
+.main h1::before { content: none !important; }
```
Comment (chrishtr) [inline www/book.css:35]: What is this rule for?

File: www/book.css (hunk 1 above)
Comment (pavpanchekha) [reply www/book.css:57]: Ah—sort of bizarrely the "Part XXX" headers were bigger than the "Introduction" and "Conclusion" headers because the latter are inside a <section>.

File: www/book.css (hunk 2 above)
Comment (pavpanchekha) [reply www/book.css:35]: On the main page (`index.html`), we don't want to auto-number the headings.

Comment (chrishtr) [reply www/book.css:35]: Please add a comment saying that then.

File: www/book.css (hunk 1 above)
Comment (chrishtr) [reply www/book.css:57]: Can you add a comment explaining?

File: www/book.css [hunk 3]
```diff
@@ -31,9 +31,17 @@ header, h1, strong {
 }
 }
 h1 { font-size: 105%; margin: 1.5em 0 0; }
+
+/* Auto-number section headings, but not on the main page. */
```
Comment (chrishtr) [inline www/book.css:35]: Idea: how about we include the section numbers on the homepage and implement it with counter-reset?

Comment (pavpanchekha) [reply www/book.css:35]: I'm a bit confused. When you say "section numbers", do you mean:

- Auto-number the parts?
- Auto-number the chapters?
- Auto-number the sections in each chapter?

Comment (chrishtr) [reply www/book.css:35]: I  mean sections within chapters

Comment (pavpanchekha) [reply www/book.css:35]: Right now the homepage doesn't show the sections. You want to add them? I guess that would look more like the printed book, not a bad idea. But perhaps a separate PR?

Comment (chrishtr) [reply www/book.css:35]: ok
